### PR TITLE
Added Forced platform

### PR DESCRIPTION
**Pull request by maine98**
https://user-images.githubusercontent.com/46499668/103581972-45293700-4edd-11eb-8e0e-a499e0427492.mp4

[FIX]
   - Added forced platform settings

[ADD]
   - Added CustomControlsPage and custom controls component

### CODE REVIEW

File: example/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (jhomlala) [inline example/.vscode/launch.json:1]: Please remove this file.

File: example/lib/pages/custom_controller_page.dart [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import 'package:better_player_example/pages/custom_controls/CustomControlsWidget.dart';
+import 'package:flutter/material.dart';
+
+class CustomControllerPage extends StatefulWidget {
+  @override
+  _CustomControllerPageState createState() => _CustomControllerPageState();
+}
+
+class _CustomControllerPageState extends State<CustomControllerPage> {
+  BetterPlayerController _betterPlayerControllerMaterial;
```
Comment (jhomlala) [inline example/lib/pages/custom_controller_page.dart:11]: Can you try do it with only one controller? User can change player style with 3 buttons.

File: example/lib/pages/custom_controller_page.dart [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            BetterPlayer(
+              controller: _betterPlayerControllerCustom,
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void _initPlayers() {
```
Comment (jhomlala) [inline example/lib/pages/custom_controller_page.dart:69]: Please move it above build method.

File: example/lib/pages/custom_controller_page.dart [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    String url =
+        'https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerBlazes.mp4';
```
Comment (jhomlala) [inline example/lib/pages/custom_controller_page.dart:71]: Please use urls from constants class.

File: example/lib/pages/welcome_page.dart [hunk 5]
```diff
@@ -77,6 +78,9 @@ class _WelcomePageState extends State<WelcomePage> {
       _buildExampleElementWidget("Controls configuration", () {
         _navigateToPage(ControlsConfigurationPage());
       }),
+      _buildExampleElementWidget("Custom Controls Player", () {
```
Comment (jhomlala) [inline example/lib/pages/welcome_page.dart:81]: Move it to the bottom of the list. In the future I will re-order list of examples based on priority.

File: lib/src/configuration/better_player_configuration.dart [hunk 6]
```diff
@@ -1,14 +1,8 @@
-// Flutter imports:
```
Comment (jhomlala) [inline lib/src/configuration/better_player_configuration.dart:1]: Revert these changes please. I don't lnow what's the purpose of this change.

File: lib/src/controls/better_player_controls_configuration.dart [hunk 7]
```diff
@@ -76,7 +77,10 @@ class BetterPlayerControlsConfiguration {
   final Duration controlsHideTime;
 
   ///Custom controls, it will override Material/Cupertino controls
-  final Widget customControls;
+  final Widget Function(BetterPlayerController controller) customControls;
```
Comment (jhomlala) [inline lib/src/controls/better_player_controls_configuration.dart:80]: final Widget Function(BetterPlayerController) customControlsBuilder seems to be better name for me.

File: lib/src/controls/better_player_controls_configuration.dart [hunk 8]
```diff
@@ -76,7 +77,10 @@ class BetterPlayerControlsConfiguration {
...
+
+  ///Flag to use a forced platform
+  final PlayerPlatform playerPlatform;
```
Comment (jhomlala) [inline lib/src/controls/better_player_controls_configuration.dart:83]: Basically I think that we shouldn't configure player theme based on platform rather than platform theme.

File: lib/src/controls/better_player_controls_configuration.dart [hunk 9]
```diff
@@ -193,3 +198,5 @@ class BetterPlayerControlsConfiguration {
         enableProgressText: true);
   }
 }
+
+enum PlayerPlatform { IOS, ANDROID }
```
Comment (jhomlala) [inline lib/src/controls/better_player_controls_configuration.dart:202]: Please change PlayerPlatform to PlayerTheme with values: `Material,` `Cupertino`. Please move it to another class.

File: lib/src/core/better_player_with_controls.dart [hunk 10]
```diff
@@ -158,18 +158,40 @@ class _BetterPlayerWithControlsState extends State<BetterPlayerWithControls> {
...
-      if (Platform.isAndroid) {
-        return BetterPlayerMaterialControls(
-          onControlsVisibilityChanged: onControlsVisibilityChanged,
-          controlsConfiguration: controlsConfiguration,
-        );
+    if (controlsConfiguration.showControls) {
+      if (controlsConfiguration.customControls == null &&
+          controlsConfiguration.playerPlatform == null) {
+        if (Platform.isAndroid) {
+          return BetterPlayerMaterialControls(
```
Comment (jhomlala) [inline lib/src/core/better_player_with_controls.dart:165]: Please create 2 methods which will create Material and Cupertino controls when requested. I believe we have code duplication here.

File: lib/src/controls/better_player_controls_configuration.dart (hunk 8 above)
Comment (jhomlala) [reply lib/src/controls/better_player_controls_configuration.dart:83]: Also please set default theme based on platform i.e.:
PlayerTheme.material for Android
PlayerTheme.cupertino for iOS

File: lib/src/core/better_player_with_controls.dart (hunk 10 above)
Comment (jhomlala) [reply lib/src/core/better_player_with_controls.dart:165]: Also based on comment above, when we set PlayerTheme default value for given platform, we don't need to react here. So we dont  this if block:
``` if (controlsConfiguration.showControls) {
      if (controlsConfiguration.customControls == null &&
          controlsConfiguration.playerPlatform == null) {
        if (Platform.isAndroid) {
...
```

So we need only this if block:
```
if (showControls){
	if (customControls){
		return buildCustomControls();
	} else if (platformTheme == material){
		return buildMaterialControls();
	} else if (platformTheme == cupertino){
		return buildCupertinoControls();
	}
} 
```

**Comment by jhomlala** [comment]
Thank you for PR. This seems to be useful, but you need to fix some code.

**Comment by maine98** [comment]
https://user-images.githubusercontent.com/46499668/103800993-63ae4000-504d-11eb-8540-1f57fb4db7e1.mp4




### [REMOVE]
   - Added .vscode/ from .gitignore
### [ADD]
   - Added enum PlayerTheme in another file and edded export inside betterplayer.dart
   - Added on enum PlayerTheme value custom for use a custom theme for controls player
### [REFACTOR]
   - Refactor page and file "custom_controls_page" in "change_player_theme_page"
   - Refactor file "CustomControlsWidget.dart" in "custom_controls_widget.dart"
### [FIX]
   - Fixed page "custom_controls_page" for using a only one controller for betterPlayer
   - Cleanup and reformat code
### [REVERT]
   - Revert modify inside "better_player_configurations.dart"

**Comment by jhomlala** [comment]
LGTM, Thanks.
